browser/profiles/brave_renderer_updater.cc

```diff
@@ -107,10 +122,12 @@ void BraveRendererUpdater::UpdateRenderer(
            brave_wallet::mojom::DefaultWallet::BraveWalletPreferExtension ||
        default_ethereum_wallet ==
            brave_wallet::mojom::DefaultWallet::BraveWallet) &&
-      is_wallet_allowed_for_context_ && brave_wallet::IsDappsSupportEnabled();
+      is_wallet_allowed_for_context_ && brave_wallet::IsDappsSupportEnabled() &&
+      !should_ignore_brave_wallet;
   bool allow_overwrite_window_ethereum_provider =
       default_ethereum_wallet ==
-      brave_wallet::mojom::DefaultWallet::BraveWalletPreferExtension;
+          brave_wallet::mojom::DefaultWallet::BraveWalletPreferExtension &&
+      !should_ignore_brave_wallet;
```

Comment:
This is not needed. It will only be used when wallet provider is injected which is based on `brave_use_native_ethereum_wallet`